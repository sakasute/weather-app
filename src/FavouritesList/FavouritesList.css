.favourites {
  list-style: none;
  padding: 0;
}

.favourites__item {
  cursor: pointer;
  margin: 0.25rem 0;
  padding: 0.25rem;
  text-transform: uppercase;
  font-weight: bold;
}

.favourites__item:hover {
  background-color: hsl(221, 100%, 95%);
}

.favourites__item--selected {
  background-color: hsl(221, 100%, 90%);
}
