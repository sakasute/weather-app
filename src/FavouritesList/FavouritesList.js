import React from "react";

export default function FavouritesList({ favourites, selectCity }) {
  const favouriteLis = Object.values(favourites).map(favourite => {
    const { id, name, country } = favourite;
    return (
      <li
        key={id}
        onClick={() => selectCity(id, name, country)}
      >{`${name}, ${country}`}</li>
    );
  });

  return <ul>{favouriteLis}</ul>;
}
