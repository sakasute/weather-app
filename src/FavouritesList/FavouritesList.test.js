import React from "react";
import FavouritesList from "./FavouritesList";
import renderer from "react-test-renderer";

it("Favourites list renders correctly", () => {
  const favourites = {
    "633679": {
      id: 633679,
      name: "Turku",
      country: "FI"
    },
    "658225": {
      id: 658225,
      name: "Helsinki",
      country: "FI"
    }
  };

  const selectCity = (id, name, country) => {
    return;
  };

  const selectedCityId = "633679";

  const tree = renderer
    .create(
      <FavouritesList
        favourites={favourites}
        selectCity={selectCity}
        selectedCityId={selectedCityId}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
