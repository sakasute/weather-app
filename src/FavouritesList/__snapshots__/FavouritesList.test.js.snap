// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourites list renders correctly 1`] = `
<ul
  className="favourites"
>
  <li
    className="favourites__item"
    onClick={[Function]}
  >
    Helsinki, FI
  </li>
  <li
    className="favourites__item"
    onClick={[Function]}
  >
    Turku, FI
  </li>
</ul>
`;
